internal/configs/virtualserver.go

```diff
@@ -756,7 +760,7 @@ func createUpstreamsForPlus(virtualServerEx *VirtualServerEx, baseCfgParams *Con
 		upstreamName := upstreamNamer.GetNameForUpstream(u.Name)
 		upstreamNamespace := virtualServerEx.VirtualServer.Namespace
 
-		endpointsKey := GenerateEndpointsKey(upstreamNamespace, u.Service, u.Port)
+		endpointsKey := GenerateEndpointsKey(upstreamNamespace, u.Service, u.Subselector, u.Port)
```

Comment:
I suggest extending the corresponding to this function test to ensure the dynamic reconfiguration works. You can extend `TestCreateUpstreamServersForPlus` with an upstream with a subselector  (and a corresponding path and endpoints) in the VS and an upstream with a subselector (and a corresponding path and endpoints)in the VSR.

Reply:
Good suggestion. Done.